### PR TITLE
WT-6763 Fix freeing update on the chain when insert fail after inserting to the update chain

### PR DESCRIPTION
**Pull request by quchenhao**
@lukech This does not fix the problem completely. The test is running out of memory. We need to move the unit test long to a larger machine.

### CODE REVIEW

**Comment by quchenhao** [comment]
@bvpvamsikrishna I don't know how to write a test for this as it requires to inject failures after we have inserted the update to the update chain.

File: src/btree/row_modify.c [hunk 1]
```diff
@@ -206,6 +206,8 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
           session, page, cbt->ins_head, cbt->ins_stack, &ins, ins_size, skipdepth, exclusive));
     }
 
+    inserted_to_update_chain = true;
```
Comment (bvpvamsikrishna) [inline src/btree/row_modify.c:209]: This logic ensures to free the update only if it is added to update chain. How about we have allocated update but not successful in adding it to update chain, for eg, __wt_txn_modify at line 110 returns error, in that case we will not free the update, which could result in memory leak. Did I miss something ?

Comment (quchenhao) [reply src/btree/row_modify.c:209]: No, it's the opposite. This is to ensure we only free the update if it is failed to be added to the update chain.

Comment (quchenhao) [reply src/btree/row_modify.c:209]: So if we fail before this point. Update will be freed.

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm.
Add the testing method you used to the Jira ticket.

**Comment by quchenhao** [comment]
> lgtm.
> Add the testing method you used to the Jira ticket.

already did

**Pull request merged**